### PR TITLE
[nat64] add build flags for NAT64 translator

### PR DESCRIPTION
**Pull request by erjiaqing**

### CODE REVIEW

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -44,6 +44,8 @@ option(OTBR_BORDER_ROUTING "Enable Border Routing Manager" OFF)
 
 option(OTBR_BORDER_ROUTING_NAT64 "Enable NAT64 support in Border Routing Manager" OFF)
 
+option(OTBR_NAT64_TRANSLATOR "Enable NAT64 support in Border Routing Manager" OFF)
```
Comment (xuyirio) [inline etc/cmake/options.cmake:47]: The description is incorrect?

**Review by xuyirio — APPROVED** [review]

**Review by abtink — APPROVED** [review]
👍

File: CMakeLists.txt [hunk 2]
```diff
@@ -40,6 +40,7 @@ set(OTBR_NAME "${OTBR_VENDOR_NAME}_${OTBR_PRODUCT_NAME}" CACHE STRING "The packa
 set(OTBR_MESHCOP_SERVICE_INSTANCE_NAME "${OTBR_VENDOR_NAME} ${OTBR_PRODUCT_NAME}" CACHE STRING "The OTBR MeshCoP service instance name")
 set(OTBR_MDNS "avahi" CACHE STRING "mDNS publisher provider")
 set(OTBR_SYSLOG_FACILITY_ID LOG_USER CACHE STRING "Syslog logging facility")
+set(OTBR_NAT64_CIDR "192.168.64.0/24" CACHE STRING "NAT64 CIDR for OpenThread NAT64")
```
Comment (bukepo) [inline CMakeLists.txt:43]: I think users can directly set OT_POSIX_NAT64_CIDR
```suggestion
```

File: third_party/openthread/CMakeLists.txt [hunk 3]
```diff
@@ -51,6 +51,11 @@ if (NOT OT_LOG_LEVEL)
     endif()
 endif()
 
+set(OT_NAT64_TRANSLATOR ${OTBR_NAT64_TRANSLATOR} CACHE BOOL "enable NAT64 translator" FORCE)
+if (OTBR_NAT64_CIDR)
+    set(OT_POSIX_NAT64_CIDR ${OTBR_NAT64_CIDR} CACHE STRING "set IPv4 CIDR used for NAT64" FORCE)
+endif()
```
Comment (bukepo) [inline third_party/openthread/CMakeLists.txt:57]: I  think users can directly set OT_POSIX_NAT64_CIDR
```suggestion
```

**Comment by erjiaqing** [comment]
This is not actually required since we can pass OT_NAT64 to openthread directly.

**Comment by erjiaqing** [comment]
Found this PR is closed by mistake. I previous thought we have nothing todo except passing the flags, however, this PR also implements the flags for building script (& docker)

**Review by bukepo — APPROVED** [review]

**Pull request merged**